main.tf

```
@@ -1,54 +1,30 @@
+provider "azurerm" {
+  version = "1.22.1"
```

the current version doesn't have support for api management, it needs a newer one
```suggestion
    version = "1.36.1"
```